pennylane_qulacs/qulacs_device.py

```diff
@@ -254,6 +260,34 @@ def _apply_qubit_unitary(self, op, inverse=False):
         self._circuit.add_gate(unitary_gate)
         unitary_gate.update_quantum_state(self._state)
 
+    def _apply_controlled_qubit_unitary(self, op, inverse=False):
+        """Apply controlled unitary to state"""
+        # translate op wire labels to consecutive wire labels used by the device
+        device_wires = self.map_wires(op.wires)
+        target_wires = self.map_wires(op.target_wires)
+        control_wires = self.map_wires(op.control_wires)
+        control_values = op.control_values 
+        par = op.parameters
+
+        if len(control_wires) + len(target_wires) != len(device_wires):
+            raise ValueError("len(device_wire) should be equal to len(control_wires) + len(target_wires)")
+
+        if len(par[0]) != 2 ** len(target_wires):
+            raise ValueError("Unitary matrix must be of shape (2**target_wires, 2**target_wires).")
+
+        if inverse:
+            par[0] = par[0].conj().T
```

Comment:
I believe that would cause inconsistency with the rest of the code. See [here](https://github.com/PennyLaneAI/pennylane-qulacs/blob/master/pennylane_qulacs/qulacs_device.py#L249) for example.

Reply:
Yes, this is because the other functions use `op.parameters` to obtain the gate. For most gates this is fine since they are defined by their parameters. But for `ControlledQubitUnitary` we should allow users to specify either a matrix directly or the controlled version of another operator, i.e. `qml.ControlledQubitUnitary(qml.RX(np.pi/2, 0), control_wires=(1,))`.
So I would say it is fine to have this inconsistency for now.